### PR TITLE
2676 Mobile RX Refill - rx history

### PR DESCRIPTION
**Pull request by aherzberg**
## Description of change

Adding rx refill history endpoint.

## Original issue(s)
 https://github.com/department-of-veterans-affairs/va-mobile-app/issues/2676

## Things to know about this PR

This is more of an initial MVP as we still need to discuss potential pagination, sorting parameters as well as exact data required. Right now I'm just passing through every field though based on preliminary wireframes, it looks like we may need each one.

### CODE REVIEW

**Comment by aherzberg** [comment]
Couple things in particular I'd like to get looked at

1. Can the Serializer be done in a better way?
2. I'm using custom actions for each endpoint to keep them all in one controller. Thoughts? See other preview of other endpoints in `2676-mobile-rx-refill`

**Comment by jperk51** [comment]
For the serializer if all those attributes, other than id, are named the same in `rxHistory.attributes ` you could pass that entire array through like what's done in the letters serializer so you'd end up with something like `... attributes:{rxHistory:[]} ...` or you could extract what you have into an adapter/model. That being said I don't necessarily think either of those is better than what you have, just some other options to consider.

**Comment by aherzberg** [comment]
> For the serializer if all those attributes, other than id, are named the same in `rxHistory.attributes ` you could pass that entire array through like what's done in the letters serializer so you'd end up with something like `... attributes:{rxHistory:[]} ...` or you could extract what you have into an adapter/model. That being said I don't necessarily think either of those is better than what you have, just some other options to consider.

I initially implemented it mirror how the letter serializer did it but realized that putting the array of objects inside the attributes doesn't technically follow the JSON API Spec. the spec prefers the array on the initial data level.

File: modules/mobile/config/routes.rb [hunk 1]
```diff
@@ -41,6 +41,7 @@
     post '/user/phones', to: 'phones#create'
     put '/user/phones', to: 'phones#update'
     delete '/user/phones', to: 'phones#destroy'
+    get '/rx-refill/rx-history', to: 'rx_refill#get_rx_history'
```
[the remark this answers is not in the document]
Comment (aherzberg) [reply modules/mobile/config/routes.rb]: Yes. I like that better. Updated.

File: modules/mobile/spec/request/rx_refill_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  describe 'GET /mobile/v0/rx-refill/prescriptions' do
+    context 'with a valid evss response and no failed facilities' do
+      before do
+        VCR.use_cassette('rx_refill/prescriptions/gets_a_list_of_all_prescriptions') do
+          get '/mobile/v0/rx-refill/rx-history', headers: iam_headers
+        end
+      end
+
+      it 'returns expected response' do
```
[the remark this answers is not in the document]
Comment (aherzberg) [reply modules/mobile/spec/request/rx_refill_request_spec.rb]: Wasn't familiar with :aggregate_failures. Thats a good way of handling it without ballooning the line count. Updated

**Review by thilton-oddball — APPROVED** [review]

**Review by kpethtel — APPROVED** [review]
This looks good. Two thoughts:
1) does this handle upstream failures by returning a 502? I only ask because we seem to have frequent bugs around this.
2) if you expect this to have pagination later and this will be used by the client before the pagination is implemented, I'd suggest you add the pagination skeleton now. If you don't, we'll have to version this endpoint later to add pagination. I've seen cases where the VAOS team adds hard-coded pagination options to unpaginated endpoints. We might be wise to do the same to avoid this issue.

**Review by kreek — APPROVED** [review]
LGTM 🚀

**Comment by aherzberg** [comment]
Going to temporarily close this PR and make some changes now that we have a better idea of requirements before openning it back up. Going to add pagination, handling of upstream failures, filter by status

**Pull request closed without merging**